pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -0,0 +1,170 @@
+package injection
+
+import (
+	"context"
+	goerrors "errors"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/status"
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/istio"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/version"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/ingestendpoint"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/initgeneration"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/namespace/mapper"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/timeprovider"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type reconciler struct {
+	client            client.Client
+	apiReader         client.Reader
+	dynakube          *dynatracev1beta1.DynaKube
+	istioReconciler   istio.Reconciler
+	versionReconciler version.Reconciler
+}
+
+type ReconcilerBuilder func(
+	client client.Client,
+	dynatraceClient dynatrace.Client,
+	istioClient *istio.Client,
+	apiReader client.Reader,
+	fs afero.Afero,
+	dynakube *dynatracev1beta1.DynaKube,
+) controllers.Reconciler
+
+//nolint:revive
+func NewReconciler(
+	client client.Client,
+	dynatraceClient dynatrace.Client,
+	istioClient *istio.Client,
+	apiReader client.Reader,
+	fs afero.Afero,
+	dynakube *dynatracev1beta1.DynaKube,
```

Comment:
small suggestion, so the order makes more sense (I like to have the `client` and `apiReader` to be passed in as a pair)
```suggestion
func NewReconciler(
	client client.Client,
	apiReader client.Reader,
	dynatraceClient dynatrace.Client,
	istioClient *istio.Client,
	fs afero.Afero,
	dynakube *dynatracev1beta1.DynaKube,
```